config/body/body.go

```diff
@@ -6,59 +6,50 @@ import (
 	"github.com/zclconf/go-cty/cty"
 )
 
-var _ hcl.Body = &Body{}
-
-type Attributes interface {
-	JustAllAttributes() []hcl.Attributes
-	JustAllAttributesWithName(string) []hcl.Attributes
-}
-
-type Body struct {
-	content *hcl.BodyContent
-}
-
-func New(content *hcl.BodyContent) hcl.Body {
-	return &Body{content}
-}
-
-func NewContentWithAttrName(name, value string) *hcl.BodyContent {
-	return NewContentWithAttr(&hcl.Attribute{
-		Name: name,
-		Expr: &hclsyntax.LiteralValueExpr{Val: cty.StringVal(value)},
-	})
-}
-
-func NewContentWithAttr(attr *hcl.Attribute) *hcl.BodyContent {
-	return &hcl.BodyContent{Attributes: map[string]*hcl.Attribute{
-		attr.Name: attr,
-	}}
-}
-
-func RenameAttribute(content *hcl.BodyContent, old, new string) {
-	if attr, ok := content.Attributes[old]; ok {
-		attr.Name = new
-		content.Attributes[new] = attr
-		delete(content.Attributes, old)
+func NewHCLSyntaxBodyWithAttr(name string, value cty.Value, rng hcl.Range) *hclsyntax.Body {
+	return &hclsyntax.Body{
+		Attributes: hclsyntax.Attributes{
+			name: {
+				Name: name,
+				Expr: &hclsyntax.LiteralValueExpr{Val: value, SrcRange: rng},
+			},
+		},
 	}
 }
 
-func (e *Body) Content(_ *hcl.BodySchema) (*hcl.BodyContent, hcl.Diagnostics) {
-	return e.content, nil
+func NewHCLSyntaxBodyWithStringAttr(name, value string) *hclsyntax.Body {
+	return NewHCLSyntaxBodyWithAttr(name, cty.StringVal(value), hcl.Range{})
 }
 
-func (e *Body) PartialContent(_ *hcl.BodySchema) (*hcl.BodyContent, hcl.Body, hcl.Diagnostics) {
-	return e.content, e, nil
+func MergeBodies(dest, src *hclsyntax.Body, replace bool) *hclsyntax.Body {
+	if src == dest {
+		return dest
+	}
+	for k, v := range src.Attributes {
+		if _, set := dest.Attributes[k]; replace || !set {
+			dest.Attributes[k] = v
+		}
+	}
+	for _, bl := range src.Blocks {
+		dest.Blocks = append(dest.Blocks, bl)
```

Comment:
Should the replace option apply to blocks too? (recursive)

```hcl
# dest
body {
  block "label-a" {
    attr1 = true
  }
}
# +
# src
body {
  block "label-a" {
    attr1 = false
  }
}
```

would result in two blocks instead of attr override.

```hcl
body {
  block "label-a" {
    attr1 = true
  }

  block "label-a" {
    attr1 = false
  }
}
```

Reply:
I'm not sure. At least, it's currently not necessary: apart from its own tests, `MergeBodies()` is used with `replace == true` in
* config/configload/endpoint.go (`wsBody` is a body containing a `websockets` block, but this is only called if the `proxy` block does not have a `websockets` child block in the first place):
  ```go
					proxyConfig.Remain = hclbody.MergeBodies(proxyConfig.HCLBody(), wsBody, true)
  ```

* handler/transport/backend_test.go (where the src does not contain blocks):
  ```go
		content := &hclsyntax.Body{Attributes: hclsyntax.Attributes{
			"connect_timeout": {Name: "connect_timeout", Expr: &hclsyntax.LiteralValueExpr{Val: cty.StringVal(connect)}},
			"ttfb_timeout":    {Name: "ttfb_timeout", Expr: &hclsyntax.LiteralValueExpr{Val: cty.StringVal(ttfb)}},
			"timeout":         {Name: "timeout", Expr: &hclsyntax.LiteralValueExpr{Val: cty.StringVal(timeout)}},
		}}
		return hclbody.MergeBodies(base, content, true)
  // ...
			backend := transport.NewBackend(hclbody.MergeBodies(hclContext,
				hclbody.NewHCLSyntaxBodyWithStringAttr("timeout", "1s"),
				true,
			), &transport.Config{}, nil, nullLog)
  ```